courses/credentials.py

```diff
@@ -0,0 +1,92 @@
+"""Digital courseware credentials"""
+from typing import Dict, Union
+from urllib.parse import urljoin
+
+from django.conf import settings
+from mitol.common.utils import now_in_utc
+from mitol.digitalcredentials.models import LearnerDID
+
+from courses.models import (
+    CourseRun,
+    CourseRunCertificate,
+    ProgramCertificate,
+    ProgramRun,
+)
+
+
+def build_program_run_completion(program_run: ProgramRun) -> Dict:
+    """Build a completion object for a ProgramRun"""
+    return {
+        "type": "schema:EducationalOccupationalProgram",
+        "identifier": program_run.program.text_id,
+        "name": program_run.program.title,
+        "description": program_run.program.page.description,
+        "numberOfCredits": {"value": program_run.program.page.certificate_page.CEUs},
+        "startDate": program_run.start_date,
+        "endDate": program_run.end_date,
+        "educationalCredentialAwarded": {
+            "type": "schema:EducationalOccupationalCredential",
+            "name": f"{program_run.program.title} Completion",
+            "description": program_run.program.page.description,
+        },
+    }
+
+
+def build_course_run_completion(course_run: CourseRun) -> Dict:
+    """Build a completion object for a CourseRun"""
+    return {
+        "type": "schema:Course",
+        "courseCode": course_run.course.readable_id,
+        "name": course_run.course.title,
+        "description": course_run.course.page.description,
+        "numberOfCredits": {"value": course_run.course.page.certificate_page.CEUs},
```

Comment:
Same as for the program, I think we should raise an error if these fields are empty.